ERCS/erc-7555.md

```diff
@@ -0,0 +1,236 @@
+
+### Definitions
+- **Smart account** - An ERC-4337 compliant smart contract account that has a modular architecture.
+- **Domain** - A string of text acting as an identification to a server or wesbite (eg: `ethereum.org` or `ABCDE12345.com.example.app`).
+- **EOA** - Accounts that are controlled by a single private key.
+- **Provider** - A third party service provider that is able to authenticate a user and produce a keypair for the user.
+
+### Redirects
+An application looking to authenticate a user must navigate the user to a given provider's URI based on the `URI Request Syntax`. The application must implement a valid redirect URI for the callback in order to receive a valid response.
+
+#### Available Routes
+- `/auth/`: The route used to authneticate a user, and request credentials.
+- `/sendTransaction/`: The route used to send a transaction payload for a user to sign. This is more of a convenient method to allow applications to do both authentication, and plugin registration within a single redirect, instead of requiring the user to perform two redirects.
+
+### Schema
+The signer type represents the specific curve used when generating the keypair. The following are a non-exhaustive list of possible keypair types:
+```=
+type SIGNER_TYPE: string = "secp256k1" | "P256" | "BLS12-381";
+```
+The `signer_key` should be returned in the did:key format, as specified by the W3C.
+The `smart_account_address` should be returned in the CAIP-10 format.
+
+#### Auth Route
+##### Request Schema
+```=
+ parameters:
+    - in: query
+      name: redirect_uri
+      schema:
+        type: string
+      description: The uri that the provider should redirect back to.
+   - in: query
+      name: chain_id
+      schema:
+        type: string
+      description: The chain_id of a given network.
+```
+##### Response Schema
+```=
+ parameters:
+    - in: query
+      name: signer_key
+      schema:
+        type: string
+      description: The public key of the user, formatted using W3C definition of did:key
+  - in: query
+      name: smart_account_address
+      schema:
+        type: string
+      description: The on-chain address for a given smart account, formatted using CAIP-10
+```
+
+##### Request Syntax
+```=
+https://<PROVIDER_URI>/auth/?
+    redirect_uri=<YOUR_REDIRECT_URI>
+    &chain_id=<CHAIN_ID>
+```
+##### Response Syntax
+```=
+https://<YOUR_REDIRECT_URI>/auth/?
+    signer_key=<SIGNER_KEY>
+    &smart_account_address=<SMART_ACCOUNT_ADDRESS>
+```
+
+#### sendTransaction Route
+##### Request Schema
+```=
+ parameters:
+    - in: query
+      name: redirect_uri
+      schema:
+        type: string
+      description: The uri that the provider should redirect back to.
+   - in: query
+      name: chain_id
+      schema:
+        type: string
+      description: The chain_id of a given network.
+   - in: query
+      name: transaction
+      schema:
+        type: string
+      description: The RLP encoded transaction that needs to be signed
```

Comment:
This doesn't support ERC4337 as you need to abi encode userOp, sign it and submit to eth_sendUserOperation.

should we have a separate route for signing/sending userOp?

Reply:
could we just change this to a userOp instead, or support userOp OR RLP encoded tx?

Reply:
```suggestion
      description: The RLP encoded transaction or [userOperation](https://github.com/ethereum/ERCs/blob/master/ERCS/erc-4337.md#useroperation) that needs to be signed
```